editor/src/components/editor/server.ts

```diff
@@ -311,3 +320,46 @@ export async function saveUserConfiguration(userConfig: UserConfiguration): Prom
     throw new Error(`server responded with ${response.status} ${response.statusText}`)
   }
 }
+
+export async function downloadGithubRepo(owner: string, repo: string): Promise<ProjectContents> {
+  const url = urljoin(UTOPIA_BACKEND, 'github', owner, repo)
+  const response = await fetch(url, {
+    method: 'GET',
+    credentials: 'include',
+    mode: MODE,
+  })
+  if (response.ok) {
+    const buffer = await response.arrayBuffer()
+    const zipFile = await jszip.loadAsync(buffer)
+    let loadedProject: ProjectContents = {}
+    let promises: Array<Promise<void>> = []
+    const loadFile = async (fileName: string, file: jszip.JSZipObject) => {
+      const shiftedFileName = fileName.replace(/[^\/]*\//, '/') // Github uses the project name and a commit hash as the root directory
+      if (file.dir) {
+        loadedProject[shiftedFileName] = directory()
+      } else {
+        const fileBuffer = await file.async('nodebuffer')
+        if (isText(shiftedFileName, fileBuffer)) {
+          const fileContents = await file.async('string')
+          loadedProject[shiftedFileName] = codeFile(fileContents, null)
+        } else {
+          // TODO Upload assets
+          loadedProject[shiftedFileName] = assetFile()
+        }
+      }
```

Comment:
I'm going to change this code to use [`fileTypeFromFileName` in `project-file-utils.ts`](https://github.com/concrete-utopia/utopia/blob/master/editor/src/core/model/project-file-utils.ts#L372) in my follow up PR, but I think we'll still have to use `isText` before blindly accepting a file as being text